### PR TITLE
Rendering: Fix user information when using render key

### PR DESCRIPTION
**Pull request by AgnesToulet**
**What this PR does / why we need it**:
This PR fixes the user information when using a render key. It was missing multiple information as:
- Teams information, that caused errors when trying to render a panel with a data source having team permissions
- User email, that's causing issues when using the ${__user.email} variable in the frontend.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
~~Please note that the bug this PR fixes is only affecting Enterprise instances using the legacy datasource guardian (so with RBAC disabled) so we could skip this fix if you see any drawback merging this. However, I think it makes sense to have the same user information using a render key or a token as it could cause other bugs in the future.~~
-> This was true for the first bug (caused by missing team information) but we discovered another bug with the email

### CODE REVIEW

File: pkg/services/contexthandler/contexthandler.go [hunk 1]
```diff
@@ -451,12 +451,14 @@ func (h *ContextHandler) initContextWithRenderAuth(reqContext *models.ReqContext
 		return true
 	}
 
-	reqContext.IsSignedIn = true
-	reqContext.SignedInUser = &models.SignedInUser{
-		OrgId:   renderUser.OrgID,
-		UserId:  renderUser.UserID,
-		OrgRole: models.RoleType(renderUser.OrgRole),
+	query := models.GetSignedInUserQuery{UserId: renderUser.UserID, OrgId: renderUser.OrgID}
+	if err := h.SQLStore.GetSignedInUserWithCacheCtx(ctx, &query); err != nil {
```
Comment (ArturWierzbicki) [inline pkg/services/contexthandler/contexthandler.go:455]: I remember one of the user queries had issues with retrieving service accounts - will this work if the `renderUser` points to a user thats actually a service account?

Comment (AgnesToulet) [reply pkg/services/contexthandler/contexthandler.go]: Tested it and it works fine with service accounts.

**Comment by AgnesToulet** [comment]
Great catch, thank you for the feedback!

I updated the PR to fallback on the old behavior when we can't retrieve the user or when the userID is set to 0 (right now it's causing an error but in the future it could retrieve a wrong user so I prefer to keep it safe).

**Review by ArturWierzbicki — APPROVED** [review]
LGTM 🚀

**Pull request merged**